c_glib/arrow-glib/local-file-system.cpp

```diff
@@ -0,0 +1,209 @@
+{
+  auto priv = GARROW_LOCAL_FILE_SYSTEM_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_LOCAL_FILE_SYSTEM_OPTIONS_USE_MMAP:
+    g_value_set_boolean(value, priv->local_file_system_options.use_mmap);
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_local_file_system_options_finalize(GObject *object)
+{
+  auto priv = GARROW_LOCAL_FILE_SYSTEM_OPTIONS_GET_PRIVATE(object);
+
+  priv->local_file_system_options.~LocalFileSystemOptions();
+
+  G_OBJECT_CLASS(garrow_local_file_system_options_parent_class)->finalize(object);
+}
+
+static void
+garrow_local_file_system_options_init(GArrowLocalFileSystemOptions *object)
+{
+  auto priv = GARROW_LOCAL_FILE_SYSTEM_OPTIONS_GET_PRIVATE(object);
+  new(&priv->local_file_system_options) arrow::fs::LocalFileSystemOptions;
+}
+
+static void
+garrow_local_file_system_options_class_init(GArrowLocalFileSystemOptionsClass *klass)
+{
+  GParamSpec *spec;
+
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->finalize     = garrow_local_file_system_options_finalize;
+  gobject_class->set_property = garrow_local_file_system_options_set_property;
+  gobject_class->get_property = garrow_local_file_system_options_get_property;
+
+  auto local_file_system_options = arrow::fs::LocalFileSystemOptions::Defaults();
+
+  /**
+   * GArrowLocalFileSystemOptions:use-mmap:
+   *
+   * Whether open_input_stream and open_input_file return a mmap'ed file,
+   * or a regular one.
+   *
+   * Since: 1.0.0
+   */
+  spec = g_param_spec_boolean("use-mmap",
+                              "Use mmap",
+                              "Whether to use mmap",
+                              local_file_system_options.use_mmap,
+                              static_cast<GParamFlags>(G_PARAM_READWRITE));
+  g_object_class_install_property(gobject_class,
+                                  PROP_LOCAL_FILE_SYSTEM_OPTIONS_USE_MMAP,
+                                  spec);
+}
+
+/**
+ * garrow_local_file_system_options_defaults:
+ *
+ * Returns: (transfer full): A newly created #GArrowLocalFileSystemOptions.
+ *
+ * Since: 1.0.0
+ */
+GArrowLocalFileSystemOptions *
+garrow_local_file_system_options_defaults(void)
+{
+  auto local_file_system_options =
+    GARROW_LOCAL_FILE_SYSTEM_OPTIONS(g_object_new(GARROW_TYPE_LOCAL_FILE_SYSTEM_OPTIONS, NULL));
+  auto priv = GARROW_LOCAL_FILE_SYSTEM_OPTIONS_GET_PRIVATE(local_file_system_options);
+
+  priv->local_file_system_options = arrow::fs::LocalFileSystemOptions::Defaults();
+
+  return local_file_system_options;
+}
+
+/* arrow::fs::LocalFileSystem */
+
+G_DEFINE_TYPE(GArrowLocalFileSystem,
+              garrow_local_file_system,
+              GARROW_TYPE_FILE_SYSTEM)
+
+static void
+garrow_local_file_system_init(GArrowLocalFileSystem *file_system)
+{
+}
+
+static void
+garrow_local_file_system_class_init(GArrowLocalFileSystemClass *klass)
+{
+}
+
+/**
+ * garrow_local_file_system_new:
```

Comment:
How about unifying `garrow_local_file_system_new()` and `garrow_local_file_system_new_with_options()` by accepting `NULL` as `options`?